routing/external_resources.rst

```diff
@@ -4,19 +4,37 @@
 How to Include External Routing Resources
 =========================================
 
-All routes are loaded via a single configuration file - usually ``app/config/routing.yml``
-(see :ref:`routing-creating-routes`). However, if you use routing annotations,
-you'll need to point the router to the controllers with the annotations.
-This can be done by "importing" directories into the routing configuration:
+Simple applications can define all their routes in a single configuration file -
+usually ``app/config/routing.yml`` (see :ref:`routing-creating-routes`).
+However, in most applications it's common to import routes definitions from
+different resources: PHP annotations in controller files, YAML or XML files
+stored in some directory, etc.
+
+This can be done by importing routing resources from the main routing file:
 
 .. configuration-block::
 
     .. code-block:: yaml
 
         # app/config/routing.yml
-        app:
+        app_file:
+            # loads routes from the given routing file stored in some bundle
+            resource: '@AcmeOtherBundle/Resources/config/routing.yml'
+
+        app_annotations:
+            # loads routes from the PHP annotations of the controllers found in that directory
             resource: '@AppBundle/Controller/'
-            type:     annotation # required to enable the Annotation reader for this resource
+            type:     annotation
+
+        app_directory:
+            # loads routes from the YAML or XML files found in that directory
+            resource: '../legacy/routing/'
+            type:     directory
+
+        app_bundle:
+            # loads routes from the YAML or XML files found in some bundle directory
+            resource: '@AppBundle/Resources/config/routing/public/'
+            type:     directory
```

Comment:
just a question. having values of this keys aligned by column is by convention or something?

Reply:
yes ... but we only do that for the docs, to improve readability. In code we never align code like this.

Reply:
:+1: